### PR TITLE
VAULT-12112: add openapi response structures for `/sys/auth/*` endpoints

### PR DESCRIPTION
**Pull request by dhuckins**
|Path|Operation|response|docs|
|----|----|----|----|
|/sys/auth|`READ`|[link](https://github.com/hashicorp/vault/blob/ea400bb64fd384b92f6c34c954c4a07e0046cb99/vault/logical_system.go#L2377)||
|/sys/auth/{path}/tune|`READ`|[link](https://github.com/hashicorp/vault/blob/ea400bb64fd384b92f6c34c954c4a07e0046cb99/vault/logical_system.go#L1590-L1646)||
|/sys/auth/{path}/tune|`UPDATE`|[link](https://github.com/hashicorp/vault/blob/ea400bb64fd384b92f6c34c954c4a07e0046cb99/vault/logical_system.go#L2120)||
|/sys/auth/{path}|`READ`|[link](https://github.com/hashicorp/vault/blob/ea400bb64fd384b92f6c34c954c4a07e0046cb99/vault/logical_system.go#L920-L978)|[docs](https://developer.hashicorp.com/vault/api-docs/system/auth)|
|/sys/auth/{path}|`UPDATE`|[link](https://github.com/hashicorp/vault/blob/ea400bb64fd384b92f6c34c954c4a07e0046cb99/vault/logical_system.go#L2621-L2635)||
|/sys/auth/{path}|`DELETE`|[link](https://github.com/hashicorp/vault/blob/ea400bb64fd384b92f6c34c954c4a07e0046cb99/vault/logical_system.go#L2754)||

tests added

### CODE REVIEW

File: vault/logical_system_paths.go [hunk 1]
```diff
@@ -1586,11 +1596,91 @@ func (b *SystemBackend) authPaths() []*framework.Path {
...
+									Required: false,
+								},
+								"audit_non_hmac_request_keys": {
+									Type:     framework.TypeCommaStringSlice,
+									Required: false,
+								},
+								"audit_non_hmac_response_keys": {
+									Type:     framework.TypeCommaStringSlice,
+									Required: false,
+								},
```
Comment (averche) [inline vault/logical_system_paths.go:1634]:
```suggestion
								},
								"listing_visibility": {
									Type:     framework.String,
									Required: false,
								},
```

**Review by averche — COMMENTED** [review]
Found one issue with the response otherwise LGTM. Could you please also update the PR title to something more descriptive

File: vault/logical_system_paths.go (hunk 1 above)
Comment (dhuckins) [reply vault/logical_system_paths.go:1634]: oh, did i miss a field?

Comment (averche) [reply vault/logical_system_paths.go:1634]: I believe so :)

Comment (dhuckins) [reply vault/logical_system_paths.go:1634]: 💩

**Comment by dhuckins** [comment]
@averche added field

**Review by averche — APPROVED** [review]
LGTM 👍

**Pull request merged**